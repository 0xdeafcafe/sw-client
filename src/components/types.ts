export type DisplayProps = {
  currentRoot: string;
  item: FilmItemProps &
    PersonItemProps &
    PlanetItemProps &
    SpeciesItemProps &
    StarshipItemProps &
    VehicleItemProps;
};

export type FilmProps = {
  item: FilmItemProps;
  isItemFavourite: boolean;
  setIsItemFavourite: (boolean: boolean) => void;
};

export type FilmItemProps = {
  characters: string[];
  created: string;
  director: string;
  edited: string;
  episode_id: number;
  opening_crawl: string;
  planets: string[];
  producer: string;
  release_date: string;
  species: string[];
  starships: string[];
  title: string;
  url: string;
  vehicles: string[];
};

export type PersonProps = {
  item: PersonItemProps;
  isItemFavourite: boolean;
  setIsItemFavourite: (boolean: boolean) => void;
};

export type PersonItemProps = {
  created: string;
  edited: string;
  birth_year: string;
  eye_color: string;
  films: string[];
  gender: string;
  hair_color: string;
  height: string;
  homeworld: string;
  name: string;
  skin_color: string;
  species: string[];
  starships: string[];
  url: string;
  vehicles: string[];
  mass: string;
};

export type PlanetProps = {
  item: PlanetItemProps;
  isItemFavourite: boolean;
  setIsItemFavourite: (boolean: boolean) => void;
};

export type PlanetItemProps = {
  name: string;
  diameter: string;
  orbital_period: string;
  population: string;
  rotation_period: string;
  surface_water: string;
  gravity: string;
  terrain: string;
  climate: string;
  films: string[];
  residents: string[];
  url: string;
};

export type SpeciesProps = {
  item: SpeciesItemProps;
  isItemFavourite: boolean;
  setIsItemFavourite: (boolean: boolean) => void;
};

export type SpeciesItemProps = {
  name: string;
  average_height: string;
  average_lifespan: string;
  classification: string;
  designation: string;
  eye_colors: string;
  hair_colors: string;
  language: string;
  skin_colors: string;
  films: string[];
  people: string[];
  url: string;
};

export type StarhsipProps = {
  item: StarshipItemProps;
  isItemFavourite: boolean;
  setIsItemFavourite: (boolean: boolean) => void;
};

export type StarshipItemProps = {
  name: string;
  model: string;
  manufacturer: string;
  cargo_capacity: string;
  hyperdrive_rating: string;
  consumables: string;
  cost_in_credits: string;
  crew: string;
  length: string;
  max_atmosphering_speed: string;
  passengers: string;
  starship_class: string;
  films: string[];
  pilots: string[];
  url: string;
};

export type VehicleProps = {
  item: VehicleItemProps;
  isItemFavourite: boolean;
  setIsItemFavourite: (boolean: boolean) => void;
};

export type VehicleItemProps = {
  name: string;
  model: string;
  manufacturer: string;
  cargo_capacity: string;
  consumables: string;
  cost_in_credits: string;
  crew: string;
  length: string;
  max_atmosphering_speed: string;
  passengers: string;
  vehicle_class: string;
  films: string[];
  pilots: string[];
  url: string;
};

export type AccordionProps = {
  item: FilmItemProps &
    PersonItemProps &
    PlanetItemProps &
    SpeciesItemProps &
    StarshipItemProps &
    VehicleItemProps;
  currentRoot: string;
};

export type ModalProps = {
  displayName: String;
  result: () => React.ReactNode;
  item: string[];
};

export type State = {
  roots: {
    isLoading: boolean;
    isBranchedRootLoading: boolean;
    payload: object;
    error: string[];
    currentRoot: StateCurrentRoot;
    branchedRoot: FilmItemProps[] &
      PersonItemProps[] &
      PlanetItemProps[] &
      SpeciesItemProps[] &
      StarshipItemProps[] &
      VehicleItemProps[];
  };
};

type StateCurrentRoot = {
  count: number;
  next: string;
  previous: string | null;
  results: FilmItemProps[] &
    PersonItemProps[] &
    PlanetItemProps[] &
    SpeciesItemProps[] &
    StarshipItemProps[] &
    VehicleItemProps[];
  k: string;
};
