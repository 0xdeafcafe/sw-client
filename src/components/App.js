import React from 'react';
import { Container, Navbar, NavbarBrand } from 'reactstrap';

const App = ({ children }) => (
	<div className={'App'}>
		<Navbar
			color={'light'}
			light
		>
			<NavbarBrand href={'/'}>{'Star Wars Client'}</NavbarBrand>
		</Navbar>

		<Container className={'pt-5'}>
			{children}
		</Container>
	</div>
);

export default App;
