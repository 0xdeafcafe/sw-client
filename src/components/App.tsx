
import React, {ReactNode} from 'react';
import { Container, Navbar, NavbarBrand } from 'reactstrap';

interface AppProps {
	children: ReactNode;
  }

const App: React.FC<AppProps> = ({ children }) => (
	<div className={'App'}>
		<Navbar
			color={'light'}
			light
		>
			<NavbarBrand href={'/'}>{'Star Wars Client'}</NavbarBrand>
		</Navbar>

		<Container className={'pt-5'}>
			{children}
		</Container>
	</div>
);

export default App;
