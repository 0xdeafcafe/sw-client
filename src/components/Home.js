import classnames from 'classnames';
import { fetchRoots } from '../actions';
import sentenceCase from 'sentence-case';
import {
	Nav,
	NavItem,
	NavLink,
	TabContent,
	TabPane,
} from 'reactstrap';
import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';

const Home = () => {
	const dispatch = useDispatch();
	const [tab, setTab] = useState(void 0);
	const roots = useSelector(state => state.roots);

	useEffect(() => {
		dispatch(fetchRoots());
	}, [dispatch]);

	const keys = Object.keys(roots.payload || {});

	return (
		<div>
			<h1>{'My little Star Wars app ðŸ‘¾'}</h1>

			{roots.payload && (
				<div className={'mt-3'}>
					<Nav tabs>
						{keys.map(k => (
							<NavItem key={k}>
								<NavLink
									className={classnames({ active: tab === k })}
									onClick={() => setTab(k)}
								>
									{sentenceCase(k)}
								</NavLink>
							</NavItem>
						))}
					</Nav>

					<TabContent activeTab={tab}>
						{keys.map(k => (
							<TabPane
								key={k}
								tabId={k}
							>
								<code>{'todo: load resource table'}</code>
							</TabPane>
						))}
					</TabContent>
				</div>
			)}
		</div>
	);
};

export default Home;
