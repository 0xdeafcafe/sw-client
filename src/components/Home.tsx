import classnames from "classnames";
import sentenceCase from "sentence-case";
import styled from "styled-components";
import { fetchIndividualRoot, fetchRoots } from "../actions/actions";
import { Nav, NavItem, NavLink, TabContent, TabPane, Button } from "reactstrap";
import CircularProgress from "@mui/material/CircularProgress";

import React, { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { ItemAccordion } from "./Accordion/Accordion";
import { FilmItemProps, PersonItemProps, PlanetItemProps, SpeciesItemProps, StarshipItemProps, VehicleItemProps } from "./types";

const SpinnerContainer = styled.div`
  display: flex;
  justify-content: center;
  padding: 50px;
`;

const StyledButtonContainer = styled.div`
  display: flex;
  justify-content: flex-end;
`

const Home = () => {
  const dispatch = useDispatch();
  const [tab, setTab] = useState<string | undefined>(void 0);

  const roots = useSelector((state: any) => state.roots);
  const isLoading = useSelector((state: any) => state.roots.isLoading);
  const currentRoot = useSelector(
    (state: any) =>
      (state &&
        state.roots &&
        state.roots.currentRoot &&
        state.roots.currentRoot.k) ||
      []
  );

  const currentRootResults = useSelector(
    (state: any) =>
      (state &&
        state.roots &&
        state.roots.currentRoot &&
        state.roots.currentRoot) ||
      []
  );

  useEffect(() => {
    dispatch(fetchRoots());
  }, [dispatch]);

  const keys = Object.keys(roots.payload || {});

  const handleTabClick = (k: string) => {
    dispatch(fetchIndividualRoot(k));
  };
  return (
    <div>
      <h1>{"My little Star Wars app ðŸ‘¾"}</h1>

      {roots.payload && (
        <div className={"mt-3"}>
          <Nav tabs>
            {keys.map((k) => (
              <NavItem key={k}>
                <NavLink
                  className={classnames({ active: tab === k })}
                  onClick={() => {
                    setTab(k);
                    handleTabClick(k);
                  }}
                >
                  {sentenceCase(k)}
                </NavLink>
              </NavItem>
            ))}
          </Nav>

          {isLoading ? (
            <SpinnerContainer>
              <CircularProgress />
            </SpinnerContainer>
          ) : (
            <>
              <TabContent activeTab={tab}>
                {keys.map((k) => (
                  <TabPane key={k} tabId={k}>
                    {currentRootResults && currentRootResults.results && currentRootResults.results.map((item: FilmItemProps & PersonItemProps & PlanetItemProps & SpeciesItemProps & StarshipItemProps & VehicleItemProps) => (
                      <ItemAccordion currentRoot={currentRoot} item={item} />
                    ))}
                  </TabPane>
                ))}
                <StyledButtonContainer>
                  {currentRootResults.previous &&<Button onClick={() => dispatch(fetchIndividualRoot(currentRootResults.previous))} >Previous</Button>}
                  {currentRootResults.next &&<Button onClick={() => dispatch(fetchIndividualRoot(currentRootResults.next))}>Next</Button>}
                </StyledButtonContainer>
              </TabContent>
            </>
          )}
        </div>
      )}
    </div>
  );
};

export default Home;
