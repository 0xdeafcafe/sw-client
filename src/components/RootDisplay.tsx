import React, {useState, useEffect} from "react";
import { RootPerson } from "./Roots/RootPerson";
import { RootPlanets } from "./Roots/RootPlanets";
import { RootSpecies } from "./Roots/RootSpecies";
import { RootStarships } from "./Roots/RootStarships";
import { RootVehicles } from "./Roots/RootVehicles";
import { RootFilms } from "./Roots/RootFilms";
import { DisplayProps } from "./types";

export const RootDisplay = ({ item, item: {url}, currentRoot }: DisplayProps) => {
  const [isItemFavourite, setIsItemFavourite] = useState(false)

  useEffect(() => {
    const localStorageItems = JSON.parse(localStorage.getItem("session") || '[]')

    if(localStorageItems.includes(url)) {
      setIsItemFavourite(true)
    }
  })

  return(
  <>
    {currentRoot === "people" && <RootPerson setIsItemFavourite={setIsItemFavourite} isItemFavourite={isItemFavourite} item={item} />}
    {currentRoot === "planets" && <RootPlanets setIsItemFavourite={setIsItemFavourite} isItemFavourite={isItemFavourite} item={item} />}
    {currentRoot === "films" && <RootFilms setIsItemFavourite={setIsItemFavourite} isItemFavourite={isItemFavourite} item={item} />}
    {currentRoot === "species" && <RootSpecies setIsItemFavourite={setIsItemFavourite} isItemFavourite={isItemFavourite} item={item} />}
    {currentRoot === "vehicles" && <RootVehicles setIsItemFavourite={setIsItemFavourite} isItemFavourite={isItemFavourite}item={item} />}
    {currentRoot === "starships" && <RootStarships setIsItemFavourite={setIsItemFavourite} isItemFavourite={isItemFavourite} item={item} />}
  </>
);
}