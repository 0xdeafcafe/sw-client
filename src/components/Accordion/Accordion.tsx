import React from "react";
import Accordion from "@mui/material/Accordion";
import AccordionSummary from "@mui/material/AccordionSummary";
import AccordionDetails from "@mui/material/AccordionDetails";
import { ExpandMore } from "@mui/icons-material";
import { RootDisplay } from "../RootDisplay";
import { AccordionProps } from "../types";

export const ItemAccordion = ({ item, currentRoot }: AccordionProps) => (
    <div>
      <Accordion style={{ background: "#F8F9FA", margin: "10px" }}>
        <AccordionSummary
          aria-controls="panel1a-content"
          expandIcon={<ExpandMore />}
          id="panel1a-header"
        >
            <p>{currentRoot === "films" ? item.title : item.name}</p>
        </AccordionSummary>
        <AccordionDetails>
            {<RootDisplay currentRoot={currentRoot} item={item} />}
        </AccordionDetails>
      </Accordion>
    </div>
  );
