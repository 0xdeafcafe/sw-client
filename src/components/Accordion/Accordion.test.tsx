
import { AccordionProps } from "components/types";
import React from "react";


import { ItemAccordion } from "./Accordion";

import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });


const props: AccordionProps = {
  currentRoot: "",
  item: {
    characters: [],
    created: "",
    director: "",
    edited: "",
    episode_id: 0,
    opening_crawl: "",
    planets: [],
    producer: "",
    release_date: "",
    species: [],
    starships: [],
    title: "",
    url: "",
    vehicles: []
  }
}

it("renders correctly when there are no items", () => {
  const wrapper = mount(<ItemAccordion {...props} />)
  expect(wrapper).toMatchSnapshot();
});
