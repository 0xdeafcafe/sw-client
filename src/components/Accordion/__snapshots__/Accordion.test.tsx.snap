// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<ItemAccordion
  currentRoot=""
  item={
    Object {
      "characters": Array [],
      "created": "",
      "director": "",
      "edited": "",
      "episode_id": 0,
      "opening_crawl": "",
      "planets": Array [],
      "producer": "",
      "release_date": "",
      "species": Array [],
      "starships": Array [],
      "title": "",
      "url": "",
      "vehicles": Array [],
    }
  }
>
  <div>
    <ForwardRef(Accordion)
      style={
        Object {
          "background": "#F8F9FA",
          "margin": "10px",
        }
      }
    >
      <MuiAccordionRoot
        className="MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters"
        ownerState={
          Object {
            "children": Array [
              <ForwardRef(AccordionSummary)
                aria-controls="panel1a-content"
                expandIcon={<UNDEFINED />}
                id="panel1a-header"
              >
                <p />
              </ForwardRef(AccordionSummary)>,
              <ForwardRef(AccordionDetails)>
                <RootDisplay
                  currentRoot=""
                  item={
                    Object {
                      "characters": Array [],
                      "created": "",
                      "director": "",
                      "edited": "",
                      "episode_id": 0,
                      "opening_crawl": "",
                      "planets": Array [],
                      "producer": "",
                      "release_date": "",
                      "species": Array [],
                      "starships": Array [],
                      "title": "",
                      "url": "",
                      "vehicles": Array [],
                    }
                  }
                />
              </ForwardRef(AccordionDetails)>,
            ],
            "disableGutters": false,
            "disabled": false,
            "expanded": false,
            "square": false,
            "style": Object {
              "background": "#F8F9FA",
              "margin": "10px",
            },
          }
        }
        square={false}
        style={
          Object {
            "background": "#F8F9FA",
            "margin": "10px",
          }
        }
      >
        <Noop />
        <ForwardRef(Paper)
          className="MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1xu1s89-MuiAccordion-root"
          square={false}
          style={
            Object {
              "background": "#F8F9FA",
              "margin": "10px",
            }
          }
        >
          <MuiPaperRoot
            as="div"
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1xu1s89-MuiAccordion-root"
            ownerState={
              Object {
                "children": Array [
                  <Context.Provider
                    value={
                      Object {
                        "disableGutters": false,
                        "disabled": false,
                        "expanded": false,
                        "toggle": [Function],
                      }
                    }
                  >
                    <ForwardRef(AccordionSummary)
                      aria-controls="panel1a-content"
                      expandIcon={<UNDEFINED />}
                      id="panel1a-header"
                    >
                      <p />
                    </ForwardRef(AccordionSummary)>
                  </Context.Provider>,
                  <ForwardRef(Collapse)
                    in={false}
                    timeout="auto"
                  >
                    <div
                      aria-labelledby="panel1a-header"
                      className="MuiAccordion-region"
                      id="panel1a-content"
                      role="region"
                    >
                      <ForwardRef(AccordionDetails)>
                        <RootDisplay
                          currentRoot=""
                          item={
                            Object {
                              "characters": Array [],
                              "created": "",
                              "director": "",
                              "edited": "",
                              "episode_id": 0,
                              "opening_crawl": "",
                              "planets": Array [],
                              "producer": "",
                              "release_date": "",
                              "species": Array [],
                              "starships": Array [],
                              "title": "",
                              "url": "",
                              "vehicles": Array [],
                            }
                          }
                        />
                      </ForwardRef(AccordionDetails)>
                    </div>
                  </ForwardRef(Collapse)>,
                ],
                "className": "MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1xu1s89-MuiAccordion-root",
                "component": "div",
                "elevation": 1,
                "square": false,
                "style": Object {
                  "background": "#F8F9FA",
                  "margin": "10px",
                },
                "variant": "elevation",
              }
            }
            style={
              Object {
                "background": "#F8F9FA",
                "margin": "10px",
              }
            }
          >
            <Noop />
            <div
              className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1elwnq4-MuiPaper-root-MuiAccordion-root"
              style={
                Object {
                  "background": "#F8F9FA",
                  "margin": "10px",
                }
              }
            >
              <ForwardRef(AccordionSummary)
                aria-controls="panel1a-content"
                expandIcon={<UNDEFINED />}
                id="panel1a-header"
                key=".0"
              >
                <MuiAccordionSummaryRoot
                  aria-controls="panel1a-content"
                  aria-expanded={false}
                  className="MuiAccordionSummary-root MuiAccordionSummary-gutters"
                  component="div"
                  disableRipple={true}
                  disabled={false}
                  focusRipple={false}
                  focusVisibleClassName="Mui-focusVisible"
                  id="panel1a-header"
                  onClick={[Function]}
                  ownerState={
                    Object {
                      "aria-controls": "panel1a-content",
                      "children": <p />,
                      "disableGutters": false,
                      "disabled": false,
                      "expandIcon": <UNDEFINED />,
                      "expanded": false,
                      "id": "panel1a-header",
                    }
                  }
                >
                  <Noop />
                  <ForwardRef(ButtonBase)
                    aria-controls="panel1a-content"
                    aria-expanded={false}
                    className="MuiAccordionSummary-root MuiAccordionSummary-gutters css-1azetr9-MuiAccordionSummary-root"
                    component="div"
                    disableRipple={true}
                    disabled={false}
                    focusRipple={false}
                    focusVisibleClassName="Mui-focusVisible"
                    id="panel1a-header"
                    onClick={[Function]}
                  >
                    <MuiButtonBaseRoot
                      aria-controls="panel1a-content"
                      aria-expanded={false}
                      as="div"
                      className="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-1azetr9-MuiAccordionSummary-root"
                      id="panel1a-header"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      ownerState={
                        Object {
                          "aria-controls": "panel1a-content",
                          "aria-expanded": false,
                          "centerRipple": false,
                          "children": Array [
                            <ForwardRef
                              className="MuiAccordionSummary-content MuiAccordionSummary-contentGutters"
                              ownerState={
                                Object {
                                  "aria-controls": "panel1a-content",
                                  "children": <p />,
                                  "disableGutters": false,
                                  "disabled": false,
                                  "expandIcon": <UNDEFINED />,
                                  "expanded": false,
                                  "id": "panel1a-header",
                                }
                              }
                            >
                              <p />
                            </ForwardRef>,
                            <ForwardRef
                              className="MuiAccordionSummary-expandIconWrapper"
                              ownerState={
                                Object {
                                  "aria-controls": "panel1a-content",
                                  "children": <p />,
                                  "disableGutters": false,
                                  "disabled": false,
                                  "expandIcon": <UNDEFINED />,
                                  "expanded": false,
                                  "id": "panel1a-header",
                                }
                              }
                            >
                              <UNDEFINED />
                            </ForwardRef>,
                          ],
                          "className": "MuiAccordionSummary-root MuiAccordionSummary-gutters css-1azetr9-MuiAccordionSummary-root",
                          "component": "div",
                          "disableRipple": true,
                          "disableTouchRipple": false,
                          "disabled": false,
                          "focusRipple": false,
                          "focusVisible": false,
                          "focusVisibleClassName": "Mui-focusVisible",
                          "id": "panel1a-header",
                          "onClick": [Function],
                          "tabIndex": 0,
                        }
                      }
                      role="button"
                      tabIndex={0}
                    >
                      <Noop />
                      <div
                        aria-controls="panel1a-content"
                        aria-expanded={false}
                        className="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
                        id="panel1a-header"
                        onBlur={[Function]}
                        onClick={[Function]}
                        onContextMenu={[Function]}
                        onDragLeave={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        role="button"
                        tabIndex={0}
                      >
                        <MuiAccordionSummaryContent
                          className="MuiAccordionSummary-content MuiAccordionSummary-contentGutters"
                          ownerState={
                            Object {
                              "aria-controls": "panel1a-content",
                              "children": <p />,
                              "disableGutters": false,
                              "disabled": false,
                              "expandIcon": <UNDEFINED />,
                              "expanded": false,
                              "id": "panel1a-header",
                            }
                          }
                        >
                          <Noop />
                          <div
                            className="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
                          >
                            <p />
                          </div>
                        </MuiAccordionSummaryContent>
                        <MuiAccordionSummaryExpandIconWrapper
                          className="MuiAccordionSummary-expandIconWrapper"
                          ownerState={
                            Object {
                              "aria-controls": "panel1a-content",
                              "children": <p />,
                              "disableGutters": false,
                              "disabled": false,
                              "expandIcon": <UNDEFINED />,
                              "expanded": false,
                              "id": "panel1a-header",
                            }
                          }
                        >
                          <Noop />
                          <div
                            className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
                          >
                            <ForwardRef(ExpandMoreIcon)>
                              <ForwardRef(SvgIcon)
                                data-testid="ExpandMoreIcon"
                              >
                                <MuiSvgIconRoot
                                  aria-hidden={true}
                                  as="svg"
                                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium"
                                  data-testid="ExpandMoreIcon"
                                  focusable="false"
                                  ownerState={
                                    Object {
                                      "children": <path
                                        d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                                      />,
                                      "color": "inherit",
                                      "component": "svg",
                                      "data-testid": "ExpandMoreIcon",
                                      "fontSize": "medium",
                                      "viewBox": "0 0 24 24",
                                    }
                                  }
                                  viewBox="0 0 24 24"
                                >
                                  <Noop />
                                  <svg
                                    aria-hidden={true}
                                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                                    data-testid="ExpandMoreIcon"
                                    focusable="false"
                                    viewBox="0 0 24 24"
                                  >
                                    <path
                                      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                                    />
                                  </svg>
                                </MuiSvgIconRoot>
                              </ForwardRef(SvgIcon)>
                            </ForwardRef(ExpandMoreIcon)>
                          </div>
                        </MuiAccordionSummaryExpandIconWrapper>
                      </div>
                    </MuiButtonBaseRoot>
                  </ForwardRef(ButtonBase)>
                </MuiAccordionSummaryRoot>
              </ForwardRef(AccordionSummary)>
              <ForwardRef(Collapse)
                in={false}
                timeout="auto"
              >
                <Transition
                  addEndListener={[Function]}
                  appear={false}
                  enter={true}
                  exit={true}
                  in={false}
                  mountOnEnter={false}
                  nodeRef={
                    Object {
                      "current": <div
                        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
                        style="min-height: 0px;"
                      >
                        <div
                          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                        >
                          <div
                            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                          >
                            <div
                              aria-labelledby="panel1a-header"
                              class="MuiAccordion-region"
                              id="panel1a-content"
                              role="region"
                            >
                              <div
                                class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                              />
                            </div>
                          </div>
                        </div>
                      </div>,
                    }
                  }
                  onEnter={[Function]}
                  onEntered={[Function]}
                  onEntering={[Function]}
                  onExit={[Function]}
                  onExited={[Function]}
                  onExiting={[Function]}
                  timeout={null}
                  unmountOnExit={false}
                >
                  <MuiCollapseRoot
                    className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden"
                    ownerState={
                      Object {
                        "children": <div
                          aria-labelledby="panel1a-header"
                          className="MuiAccordion-region"
                          id="panel1a-content"
                          role="region"
                        >
                          <ForwardRef(AccordionDetails)>
                            <RootDisplay
                              currentRoot=""
                              item={
                                Object {
                                  "characters": Array [],
                                  "created": "",
                                  "director": "",
                                  "edited": "",
                                  "episode_id": 0,
                                  "opening_crawl": "",
                                  "planets": Array [],
                                  "producer": "",
                                  "release_date": "",
                                  "species": Array [],
                                  "starships": Array [],
                                  "title": "",
                                  "url": "",
                                  "vehicles": Array [],
                                }
                              }
                            />
                          </ForwardRef(AccordionDetails)>
                        </div>,
                        "collapsedSize": "0px",
                        "in": false,
                        "orientation": "vertical",
                        "state": "exited",
                        "timeout": "auto",
                      }
                    }
                    style={
                      Object {
                        "minHeight": "0px",
                      }
                    }
                  >
                    <Noop />
                    <div
                      className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
                      style={
                        Object {
                          "minHeight": "0px",
                        }
                      }
                    >
                      <MuiCollapseWrapper
                        className="MuiCollapse-wrapper MuiCollapse-vertical"
                        ownerState={
                          Object {
                            "children": <div
                              aria-labelledby="panel1a-header"
                              className="MuiAccordion-region"
                              id="panel1a-content"
                              role="region"
                            >
                              <ForwardRef(AccordionDetails)>
                                <RootDisplay
                                  currentRoot=""
                                  item={
                                    Object {
                                      "characters": Array [],
                                      "created": "",
                                      "director": "",
                                      "edited": "",
                                      "episode_id": 0,
                                      "opening_crawl": "",
                                      "planets": Array [],
                                      "producer": "",
                                      "release_date": "",
                                      "species": Array [],
                                      "starships": Array [],
                                      "title": "",
                                      "url": "",
                                      "vehicles": Array [],
                                    }
                                  }
                                />
                              </ForwardRef(AccordionDetails)>
                            </div>,
                            "collapsedSize": "0px",
                            "in": false,
                            "orientation": "vertical",
                            "state": "exited",
                            "timeout": "auto",
                          }
                        }
                      >
                        <Noop />
                        <div
                          className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                        >
                          <MuiCollapseWrapperInner
                            className="MuiCollapse-wrapperInner MuiCollapse-vertical"
                            ownerState={
                              Object {
                                "children": <div
                                  aria-labelledby="panel1a-header"
                                  className="MuiAccordion-region"
                                  id="panel1a-content"
                                  role="region"
                                >
                                  <ForwardRef(AccordionDetails)>
                                    <RootDisplay
                                      currentRoot=""
                                      item={
                                        Object {
                                          "characters": Array [],
                                          "created": "",
                                          "director": "",
                                          "edited": "",
                                          "episode_id": 0,
                                          "opening_crawl": "",
                                          "planets": Array [],
                                          "producer": "",
                                          "release_date": "",
                                          "species": Array [],
                                          "starships": Array [],
                                          "title": "",
                                          "url": "",
                                          "vehicles": Array [],
                                        }
                                      }
                                    />
                                  </ForwardRef(AccordionDetails)>
                                </div>,
                                "collapsedSize": "0px",
                                "in": false,
                                "orientation": "vertical",
                                "state": "exited",
                                "timeout": "auto",
                              }
                            }
                          >
                            <Noop />
                            <div
                              className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                            >
                              <div
                                aria-labelledby="panel1a-header"
                                className="MuiAccordion-region"
                                id="panel1a-content"
                                role="region"
                              >
                                <ForwardRef(AccordionDetails)
                                  key=".1"
                                >
                                  <MuiAccordionDetailsRoot
                                    className="MuiAccordionDetails-root"
                                    ownerState={
                                      Object {
                                        "children": <RootDisplay
                                          currentRoot=""
                                          item={
                                            Object {
                                              "characters": Array [],
                                              "created": "",
                                              "director": "",
                                              "edited": "",
                                              "episode_id": 0,
                                              "opening_crawl": "",
                                              "planets": Array [],
                                              "producer": "",
                                              "release_date": "",
                                              "species": Array [],
                                              "starships": Array [],
                                              "title": "",
                                              "url": "",
                                              "vehicles": Array [],
                                            }
                                          }
                                        />,
                                      }
                                    }
                                  >
                                    <Noop />
                                    <div
                                      className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                                    >
                                      <RootDisplay
                                        currentRoot=""
                                        item={
                                          Object {
                                            "characters": Array [],
                                            "created": "",
                                            "director": "",
                                            "edited": "",
                                            "episode_id": 0,
                                            "opening_crawl": "",
                                            "planets": Array [],
                                            "producer": "",
                                            "release_date": "",
                                            "species": Array [],
                                            "starships": Array [],
                                            "title": "",
                                            "url": "",
                                            "vehicles": Array [],
                                          }
                                        }
                                      />
                                    </div>
                                  </MuiAccordionDetailsRoot>
                                </ForwardRef(AccordionDetails)>
                              </div>
                            </div>
                          </MuiCollapseWrapperInner>
                        </div>
                      </MuiCollapseWrapper>
                    </div>
                  </MuiCollapseRoot>
                </Transition>
              </ForwardRef(Collapse)>
            </div>
          </MuiPaperRoot>
        </ForwardRef(Paper)>
      </MuiAccordionRoot>
    </ForwardRef(Accordion)>
  </div>
</ItemAccordion>
`;
